config/routes.rb

```diff
@@ -3,6 +3,9 @@
   namespace :api do
     namespace :v1 do
       resources :trails, defaults: { format: 'json' }
+      get "/trails/search" => "trails#search", defaults: { format: 'json'}
```

Comment:
Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.